src/etc/kate/rust.xml

```diff
@@ -249,14 +252,30 @@
 			<DetectChar char="=" attribute="Normal Text" context="#pop"/>
 			<DetectChar char="&lt;" attribute="Normal Text" context="#pop"/>
 		</context>
-		<context attribute="String" lineEndContext="#pop" name="String">
-			<LineContinue attribute="String" context="#stay"/>
-			<DetectChar char="\" attribute="CharEscape" context="CharEscape"/>
-			<DetectChar attribute="String" context="#pop" char="&quot;"/>
+                <!-- Rustc allows strings to extend over multiple lines, and the
+                only thing a backshash at end-of-line does is remove the whitespace. -->
+                <context attribute="String" lineEndContext="#stay" name="String">
+                        <DetectChar char="\" attribute="CharEscape" context="CharEscape"/>
+                        <DetectChar attribute="String" context="StringEndGuard" char="&quot;"/>
+                </context>
+		<context attribute="String" lineEndContext="#stay" name="RawString">
+			<DetectChar attribute="String" context="StringEndGuard" char="&quot;"/>
 		</context>
+                <!-- These rules are't complete: they won't match r###"abc"### -->
+                <context attribute="String" lineEndContext="#stay" name="RawHashed1">
+                        <Detect2Chars attribute="String" context="StringEndGuard" char="&quot;" char1="#"/>
+                </context>
+                <context attribute="String" lineEndContext="#stay" name="RawHashed2">
+                        <StringDetect attribute="String" context="StringEndGuard" String="&quot;##"/>
+                </context>
+                <context name="StringEndGuard" attribute="Error" fallthrough="true" fallthroughContext="#pop#pop">
+                  <!-- Need a word break. This rule highlights (some) wrong characters following
+                    the end of a string. -->
+                  <RegExpr String="[#a-zA-Z0-9_&quot;&apos;]" attribute="Error"/>
```

Comment:
Nothing is an error after a string ends (which includes the right number of trailing `#` for raw strings).